### PR TITLE
Auto multi trainer

### PR DESCRIPTION
**Pull request by agsedova**

### CODE REVIEW

File: knodle/trainer/auto_multi_trainer.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import numpy as np
+from torch.utils.data import TensorDataset
+
+from knodle.trainer import AutoTrainer
+from knodle.trainer.utils import log_section
+
+
+logger = logging.getLogger(__name__)
+
+
```
Comment (AndSt) [inline knodle/trainer/auto_multi_trainer.py:14]: Switch to MultiTrainer? :D

File: knodle/trainer/wscrossweigh/wscrossweigh.py [hunk 2]
```diff
@@ -77,7 +79,9 @@ def train(
             input_info_labels_to_tensordataset(self.model_input_x, sample_weights.cpu().detach().numpy(), train_labels)
         )
 
-        self._train_loop(train_loader, use_sample_weights=True, draw_plot=True)
+        self._train_loop(train_loader, use_sample_weights=True, draw_plot=self.trainer_config.draw_plot)
```
Comment (AndSt) [inline knodle/trainer/wscrossweigh/wscrossweigh.py:82]: Are you sure this works? Never seen object access for a parameter

File: examples/trainer/simple_auto_trainer/auto_trainer_multiple_tutorial.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+# Define some functions
+from knodle.trainer.auto_multi_trainer import AutoMultiTrainer
+from knodle.trainer.wscrossweigh.config import WSCrossWeighConfig
+
+
+def np_array_to_tensor_dataset(x: np.ndarray) -> TensorDataset:
+    """
+
+    :rtype: object
```
Comment (AndSt) [inline examples/trainer/simple_auto_trainer/auto_trainer_multiple_tutorial.py:32]: Delete maybe

File: examples/trainer/simple_auto_trainer/auto_trainer_multiple_tutorial.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+# get test labels
+y_test = np_array_to_tensor_dataset(df_test['label'].values)
+
+# initilize model
+logreg_model = LogisticRegressionModel(X_train_tfidf.shape[1], 2)
+
+configs = [
+    MajorityConfig(optimizer=AdamW, lr=1e-4, batch_size=16, epochs=3),
+    kNNConfig(optimizer=AdamW, k=2, lr=1e-4, batch_size=32, epochs=2),
```
Comment (AndSt) [inline examples/trainer/simple_auto_trainer/auto_trainer_multiple_tutorial.py:89]: Dang why did we do knnConfig not capital lettered? 😅

File: examples/trainer/simple_auto_trainer/auto_trainer_tutorial.py [hunk 5]
```diff
@@ -38,7 +38,7 @@ def np_array_to_tensor_dataset(x: np.ndarray) -> TensorDataset:
 
 
 # Define constants
-imdb_data_dir = os.path.join(os.getcwd(), "data", "imdb")
+imdb_data_dir = os.path.join(os.getcwd(), "datasets", "spouse")
```
Comment (AndSt) [inline examples/trainer/simple_auto_trainer/auto_trainer_tutorial.py:41]: Is this changed in the Jupyter notebook as well?

File: knodle/trainer/baseline/majority.py [hunk 6]
```diff
@@ -67,3 +67,5 @@ def train(
         feature_label_dataloader = self._make_dataloader(feature_label_dataset)
 
         self._train_loop(feature_label_dataloader)
+
+        return self
```
Comment (AndSt) [inline knodle/trainer/baseline/majority.py:71]: I would rather not have that, as it's inconsistent with Hugging Face

File: knodle/trainer/auto_multi_trainer.py [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+            kwargs["trainer_config"] = configs[i]
+            self.trainer.append(AutoTrainer(n, **kwargs))
+
+    def train(
+            self,
+            model_input_x: TensorDataset = None, rule_matches_z: np.ndarray = None,
+            dev_model_input_x: TensorDataset = None, dev_gold_labels_y: TensorDataset = None
+    ):
+        all_trainers = copy.deepcopy(self.trainer)
+        trained_trainers = []
```
Comment (AndSt) [inline knodle/trainer/auto_multi_trainer.py:37]: Why do you need this copy?

**Review by AndSt — APPROVED** [review]

File: examples/trainer/simple_auto_trainer/auto_trainer_multiple_tutorial.py (hunk 4 above)
Comment (agsedova) [reply examples/trainer/simple_auto_trainer/auto_trainer_multiple_tutorial.py:85]: and kNNAggregationTrainer is capitalized as well haha

**Pull request merged**